spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/JavaConfigFixedReplyQueueTests.java

```diff
@@ -129,6 +197,19 @@ public SimpleMessageListenerContainer serviceListenerContainer() {
 		}
 
 		/**
+		 * @return The reply listener container - the rabbit template with the wrong queue is the listener.
+		 */
+		@Bean
+		public SimpleMessageListenerContainer replyListenerContainerWrongQueue() {
+			SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
+			container.setConnectionFactory(rabbitConnectionFactory());
+			container.setQueues(replyQueue());
+			container.setMessageListener(fixedReplyQRabbitTemplateWrongQueue());
```

Comment:
I don't think we should replace/warn; failing to start is more appropriate.

However, I agree that we should also fail to start if there is more than one queue; I will polish.